### PR TITLE
Adjust wait_for_transaction_acceptance for slow indexer

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Adjust wait_for_transaction_acceptance for slow indexer

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

`cargo ci-tangle-test`

### CODE REVIEW

File: sdk/src/client/api/wait_for_tx_acceptance.rs [hunk 1]
```diff
@@ -30,6 +33,41 @@ impl Client {
...
                     match transaction_metadata.transaction_state {
                         TransactionState::Accepted | TransactionState::Committed | TransactionState::Finalized => {
+                            let slot_index = self.get_slot_index().await?;
+                            let protocol_parameters = self.get_protocol_parameters().await?;
+                            if let Ok(output) = self.get_output(&OutputId::new(*transaction_id, 0)).await {
+                                if let Some(required_address) = output
+                                    .output
+                                    .required_address(slot_index, protocol_parameters.committable_age_range())?
+                                {
+                                    // Even thought the output was created already, the indexer might take some time
```
Comment (DaughterOfMars) [inline sdk/src/client/api/wait_for_tx_acceptance.rs]: u thought

**Review by DaughterOfMars — COMMENTED** [review]
I need an explanation pls

File: sdk/src/client/api/wait_for_tx_acceptance.rs (hunk 1 above)
Comment (Thoralf-M) [inline sdk/src/client/api/wait_for_tx_acceptance.rs]:
```suggestion
                                    // Even though the output was created already, the indexer might take some time
```

Comment (Thoralf-M) [reply sdk/src/client/api/wait_for_tx_acceptance.rs]: Thanks :D

**Comment by Thoralf-M** [comment]
@DaughterOfMars in https://github.com/iotaledger/iota-sdk/pull/2209 the private tangle tests fail because sometimes the indexer isn't updated fast enough (output exists, but is not returned by the indexer, because it wasn't processed yet)
This means when we call `wait_for_transaction_acceptance()` and afterwards right away sync the account, then we might don't get the new outputs
That's why I added the extra logic to wait until the output id is also returned from the indexer, so when we sync it will be there. I don't like it, but at the moment I don't see any other way to handle it better. If you can think of something better, please let me know

**Review by thibault-martinez — APPROVED** [review]

**Comment by DaughterOfMars** [comment]
Shouldn't we wait for that separately somehow?

**Comment by Thoralf-M** [comment]
> Shouldn't we wait for that separately somehow?

I don't think so, wait_for_transaction_acceptance is not very useful without this
However, we could rename it to `wait_for_transaction_outputs()` or similar, if you think that's better

**Comment by DaughterOfMars** [comment]
It's just that the indexer has nothing to do with the transaction acceptance, and we don't actually use any of the info we get from it here. It feels like this is not the proper place to have this logic.

**Comment by Thoralf-M** [comment]
`wait_for_transaction_acceptance()` exists because we want to wait for the outputs to be available, which was true until recently
We still want to wait until the outputs are available, otherwise this function is useless as you can see in the failing tests

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**